lib/generation/generators/middleware/state_management/utils/middleware_utils.dart

```diff
@@ -6,38 +6,73 @@ import 'package:recase/recase.dart';
 
 class MiddlewareUtils {
   static String generateChangeNotifierClass(
-    String states,
     String defaultStateName,
     PBGenerationManager manager,
-    String defaultWidgetName, {
-    List<PBSharedParameterProp> overrideProperties,
-  }) {
+    PBIntermediateNode node,
+  ) {
     var overrideVars = ''; // Variables outside of initializer
     var overrideAttr = ''; // Attributes that will be part of initializer
+    var stateInitializers = StringBuffer();
+    var stateBuffer = StringBuffer();
 
-    overrideProperties?.forEach((property) {
-      overrideVars += 'final ${property.friendlyName};';
-      overrideAttr += 'this.${property.friendlyName}, ';
+    if (node is PBSharedMasterNode &&
+        (node.overridableProperties?.isNotEmpty ?? false)) {
+      node.overridableProperties.forEach((property) {
+        overrideVars += 'final ${property.friendlyName};';
+        overrideAttr += 'this.${property.friendlyName}, ';
+      });
+      stateBuffer.write(MiddlewareUtils.generateEmptyVariable(node));
+      stateInitializers.write(
+          '${node.name.camelCase} = ${MiddlewareUtils.generateVariableBody(node)}');
+    } else {
+      stateBuffer.write(MiddlewareUtils.generateVariable(node));
+    }
+    node?.auxiliaryData?.stateGraph?.states?.forEach((state) {
+      var variationNode = state.variation.node;
+
+      if (variationNode is PBSharedMasterNode &&
+          (variationNode.overridableProperties?.isNotEmpty ?? false)) {
+        variationNode.overridableProperties.forEach((property) {
+          overrideVars += 'final ${property.friendlyName};';
+          overrideAttr += 'this.${property.friendlyName}, ';
+        });
+        stateBuffer.write(MiddlewareUtils.generateEmptyVariable(variationNode));
+        stateInitializers.write(
+            '${variationNode.name.camelCase} = ${MiddlewareUtils.generateVariableBody(variationNode)}');
+      } else {
+        stateBuffer.write(MiddlewareUtils.generateVariable(variationNode));
+      }
     });
 
     return '''
       import 'package:flutter/material.dart';
```

Comment:
We should move this to a separate Set<String> imports; (removes duplicates automatically or you can check if imports contains the string already and output all imports at the top of a file before generating classes.  We generate too many duplicate imports in some files because of this practice.

Reply:
Good observation. Will fix!